cmd/expired-authz-purger/main.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2014 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/cactus/go-statsd-client/statsd"
+	"github.com/codegangsta/cli"
+	"github.com/jmhodges/clock"
+	"gopkg.in/gorp.v1"
+
+	"github.com/letsencrypt/boulder/cmd"
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/metrics"
+	"github.com/letsencrypt/boulder/sa"
+)
+
+type expiredAuthzPurger struct {
+	stats statsd.Statter
+	log   blog.Logger
+	clk   clock.Clock
+	db    *gorp.DbMap
+
+	batchSize int64
+}
+
+func (p *expiredAuthzPurger) setDefaults() {
+	if p.batchSize == 0 {
+		p.batchSize = 1000
+	}
+}
+
+func (p *expiredAuthzPurger) purgeAuthzs(purgeBefore time.Time) (int64, error) {
+	tx, err := p.db.Begin()
+	if err != nil {
+		return 0, err
+	}
+
+	rowsAffected := int64(0)
+
+	for {
+		result, err := tx.Exec(`
+			DELETE FROM pendingAuthorizations
+			WHERE expires < ?
+			LIMIT ?
+			`,
+			purgeBefore,
+			p.batchSize,
+		)
+		if err != nil {
+			return rowsAffected, err
+		}
+		rows, err := result.RowsAffected()
+		if err != nil {
+			return rowsAffected, err
+		}
+
+		rowsAffected += rows
+		p.log.Info(fmt.Sprintf("Progress: Deleted %d (%d) expired pending authorizations", rows, rowsAffected))
+
+		if rows < p.batchSize {
+			p.log.Info(fmt.Sprintf("Deleted a total of %d expired pending authorizations", rowsAffected))
+			return rowsAffected, nil
+		}
+	}
+}
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "expired-authz-purger"
+	app.Usage = "Purge expired pending authorizations from the database"
+	app.Version = cmd.Version()
+	app.Author = "Boulder contributors"
+	app.Email = "ca-dev@letsencrypt.org"
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:   "config",
+			Value:  "config.json",
+			EnvVar: "BOULDER_CONFIG",
+			Usage:  "Path to Boulder JSON configuration file",
+		},
+	}
+
+	app.Action = func(c *cli.Context) {
+		configFileName := c.GlobalString("config")
+
+		if configFileName == "" {
```

Comment:
Since you have a default value this seems unlikely to happen unless you call it with just `-config`.